Sources/GRPCCodeGen/Internal/Renderer/TextBasedRenderer.swift

```diff
@@ -177,10 +178,18 @@ struct TextBasedRenderer: RendererProtocol {
   }
 
   /// Renders the specified import statements.
-  func renderImports(_ imports: [ImportDescription]?) { (imports ?? []).forEach(renderImport) }
+  func renderImports(_ imports: [ImportDescription]?) {
+    for (`import`, isLast) in (imports ?? []).enumeratedWithLastMarker() {
+      renderImport(`import`, isLast: isLast)
+      if isLast {
+        writer.nextLineAppendsToLastLine()
+        writer.writeLine("\n")
```

Comment:
I don't think this is the right approach as it special cases the rendering of imports. I think adding a `CodeBlockItem` for an empty line is probably the better way to do this because it's more flexible: you can then use it in other places like between declarations, for example.